wheelbase: 0.257 #[m]
width: 0.2
length: 0.5

# box constraints
# velocity
v_max: 2 # max long vel [m/s]
v_min: 0 #0.05 # min long vel [m/s]

# steering
delta_min: -0.35  # minimum steering angle [rad]
delta_max: 0.35  # maximum steering angle [rad]

# linear acceleration
a_min: -3.5
a_max: 3.5

# lateral acceleration
alat_max: 5.0

# track parameters
track_file: 'outerloop_center_smooth.csv'
track_width_R: 0.3
track_width_L: 0.3

# overtaking parameters
distance_overtaking: 1.5
distance_mergeback: 0

# ilqr parameters
T: 2
N: 7  # the number of planning steps
max_itr: 100
# replan_dt: 0.1

d_open_loop: [-0.0146, 0.0874, -0.0507, 0.0005, -0.0332, 0.1869, 0.0095, 0.0170, -0.0583, 0.0388]

front_driver:
    # track parameters
    track_offset_right: 0.15
    track_offset_left: -0.15
    prob_switch: 0.25
    ref_vel: 0.6
    ref_accel: 2.5

    # cost
    w_ref_traj: 5
    w_accel: 0.5
    w_delta: 0.5
    q1_lat: 0
    q2_lat: 0
    q1_obs: 0
    q2_obs: 0

cruise:
    # track parameters
    track_offset: 0.15
    ref_vel: 1
    ref_accel: 2.5

    # cost
    w_ref_traj: 5
    w_accel: 0.5
    w_delta: 0.5
    q1_lat: 0
    q2_lat: 0
    q1_obs: 1
    q2_obs: 5

overtake:
    # track parameters
    track_offset: -0.18
    ref_vel: 1.25
    ref_accel: 2.5

    # cost
    w_ref_traj: 5
    w_accel: 0.5
    w_delta: 0.5
    q1_lat: 0
    q2_lat: 0
    q1_obs: 0
    q2_obs: 0